pkg/providers/amifamily/ami.go

```diff
@@ -160,10 +160,14 @@ func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, nodeTemplate *v1alp
 	for _, ssmOutput := range ssmRequirements {
 		amiID, err := p.fetchAMIsFromSSM(ctx, ssmOutput.Query)
 		if err != nil {
-			return nil, err
+			logging.FromContext(ctx).With("query", ssmOutput.Query).Warnf("discover ami from ssm failed: %v", err)
+			continue
 		}
 		amis = append(amis, AMI{AmiID: amiID, Requirements: ssmOutput.Requirements})
 	}
+	if len(amis) == 0 {
```

Comment:
I think you can let the outer caller handle this condition rather than trying to handle it within the calling function directly